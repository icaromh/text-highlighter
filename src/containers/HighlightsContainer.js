import React from 'react'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import HighlightsList from '../components/HighlightsList'
import Highlight from '../components/Highlight'
import { ShareIcon } from '../components/Icons'
import HighlightFilterContainer from './HighlightFilterContainer'

import TWITTER_URL_SHARE from '../constants'

const HighlightsListContainer = ({ highlights }) => (
  <HighlightsList>
    <HighlightFilterContainer />
    <div className="HighlightsList__wrapper">
      {highlights.map(highlight => (
        <p key={`${highlight.color}${highlight.sentence}`}>
          <Highlight data={highlight} />
          <a
            target="_blank"
            rel="noopener noreferrer"
            href={TWITTER_URL_SHARE(highlight.sentence)}
          >
            <ShareIcon />
          </a>
        </p>
      ))}
    </div>
  </HighlightsList>
)

HighlightsListContainer.propTypes = {
  highlights: PropTypes.arrayOf(PropTypes.shape({
    sentence: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired,
    head: PropTypes.number,
    tail: PropTypes.number,
  })),
}

HighlightsListContainer.defaultProps = {
  highlights: [],
}

const mapStateToProps = state => ({
  highlights: (() => state.highlights.filter(h => state.filter.includes(h.color)))(),
})

export {
  HighlightsListContainer,
}

export default connect(
  mapStateToProps,
  {},
)(HighlightsListContainer)
