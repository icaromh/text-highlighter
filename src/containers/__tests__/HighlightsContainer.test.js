import React from 'react'
import { shallow } from 'enzyme'

import { HighlightsListContainer } from '../HighlightsContainer'

describe('#HighlightFilterContainer', () => {
  test('expect to render', () => {
    const wrapper = shallow(
      <HighlightsListContainer highlights={[]} />,
    )

    expect(wrapper).toMatchSnapshot()
  })

  test('expect to list filtered colors', () => {
    const highlightsList = [
      {
        head: 20,
        tail: 80,
        sentence: 'sentence a',
        color: 'red',
      },
    ]

    const wrapper = shallow(
      <HighlightsListContainer highlights={highlightsList} />,
    )

    expect(wrapper).toMatchSnapshot()
  })
})
