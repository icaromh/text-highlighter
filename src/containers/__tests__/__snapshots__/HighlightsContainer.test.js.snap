// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#HighlightFilterContainer expect to list filtered colors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HighlightsListContainer
    highlights={
      Array [
        Object {
          "color": "red",
          "head": 20,
          "sentence": "sentence a",
          "tail": 80,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Connect(HighlightFilter) />,
        <div
          className="HighlightsList__wrapper"
        >
          <p>
            <Highlight
              data={
                Object {
                  "color": "red",
                  "head": 20,
                  "sentence": "sentence a",
                  "tail": 80,
                }
              }
            />
            <a
              href="https://twitter.com/intent/tweet?text=sentence a&tw_p=tweetbutton&via=icaromh"
              rel="noopener noreferrer"
              target="_blank"
            >
              Share
            </a>
          </p>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              <Highlight
                data={
                  Object {
                    "color": "red",
                    "head": 20,
                    "sentence": "sentence a",
                    "tail": 80,
                  }
                }
              />
              <a
                href="https://twitter.com/intent/tweet?text=sentence a&tw_p=tweetbutton&via=icaromh"
                rel="noopener noreferrer"
                target="_blank"
              >
                Share
              </a>
            </p>,
          ],
          "className": "HighlightsList__wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "redsentence a",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Highlight
                  data={
                    Object {
                      "color": "red",
                      "head": 20,
                      "sentence": "sentence a",
                      "tail": 80,
                    }
                  }
                />,
                <a
                  href="https://twitter.com/intent/tweet?text=sentence a&tw_p=tweetbutton&via=icaromh"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Share
                </a>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "data": Object {
                    "color": "red",
                    "head": 20,
                    "sentence": "sentence a",
                    "tail": 80,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Share",
                  "href": "https://twitter.com/intent/tweet?text=sentence a&tw_p=tweetbutton&via=icaromh",
                  "rel": "noopener noreferrer",
                  "target": "_blank",
                },
                "ref": null,
                "rendered": "Share",
                "type": "a",
              },
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Connect(HighlightFilter) />,
          <div
            className="HighlightsList__wrapper"
          >
            <p>
              <Highlight
                data={
                  Object {
                    "color": "red",
                    "head": 20,
                    "sentence": "sentence a",
                    "tail": 80,
                  }
                }
              />
              <a
                href="https://twitter.com/intent/tweet?text=sentence a&tw_p=tweetbutton&via=icaromh"
                rel="noopener noreferrer"
                target="_blank"
              >
                Share
              </a>
            </p>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                <Highlight
                  data={
                    Object {
                      "color": "red",
                      "head": 20,
                      "sentence": "sentence a",
                      "tail": 80,
                    }
                  }
                />
                <a
                  href="https://twitter.com/intent/tweet?text=sentence a&tw_p=tweetbutton&via=icaromh"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Share
                </a>
              </p>,
            ],
            "className": "HighlightsList__wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "redsentence a",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Highlight
                    data={
                      Object {
                        "color": "red",
                        "head": 20,
                        "sentence": "sentence a",
                        "tail": 80,
                      }
                    }
                  />,
                  <a
                    href="https://twitter.com/intent/tweet?text=sentence a&tw_p=tweetbutton&via=icaromh"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    Share
                  </a>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "data": Object {
                      "color": "red",
                      "head": 20,
                      "sentence": "sentence a",
                      "tail": 80,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Share",
                    "href": "https://twitter.com/intent/tweet?text=sentence a&tw_p=tweetbutton&via=icaromh",
                    "rel": "noopener noreferrer",
                    "target": "_blank",
                  },
                  "ref": null,
                  "rendered": "Share",
                  "type": "a",
                },
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`#HighlightFilterContainer expect to render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HighlightsListContainer
    highlights={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Connect(HighlightFilter) />,
        <div
          className="HighlightsList__wrapper"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "HighlightsList__wrapper",
        },
        "ref": null,
        "rendered": Array [],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Connect(HighlightFilter) />,
          <div
            className="HighlightsList__wrapper"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "HighlightsList__wrapper",
          },
          "ref": null,
          "rendered": Array [],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
